cmd/mailbox-controller/main.go

```diff
@@ -95,54 +91,58 @@ func main() {
 		}
 	}()
 
-	// create edgeSharedInformerFactory
-	espwRestConfig, err := espwClientOpts.ToRESTConfig()
+	// create space-aware client
+	spaceManagementConfig, err := spaceMgtOpts.ToRESTConfig()
 	if err != nil {
-		logger.Error(err, "failed to create config from flags")
+		logger.Error(err, "Failed to create space management config from flags")
 		os.Exit(3)
 	}
-
-	edgeClientset, err := edgeclientset.NewForConfig(espwRestConfig)
+	spaceclient, err := spaceclient.NewMultiSpace(ctx, spaceManagementConfig)
 	if err != nil {
-		logger.Error(err, "failed to create clientset for view of edge exports")
-		os.Exit(6)
+		logger.Error(err, "Failed to create space-aware client")
+		os.Exit(4)
 	}
+	spaceProviderNs := spacemanager.ProviderNS(spaceProvider)
 
-	edgeSharedInformerFactory := edgeinformers.NewSharedScopedInformerFactoryWithOptions(edgeClientset, resyncPeriod)
-	syncTargetClusterPreInformer := edgeSharedInformerFactory.Edge().V2alpha1().SyncTargets()
+	coreRestConfig, err := spaceclient.ConfigForSpace(coreSpace, spaceProviderNs)
```

Comment:
This could have been left as `espwRestConfig`, right? A change to `kcsRestConfig` would be OK. We call this particular space role "the KubeStellar Core Space (KCS)". Let's be consistent.

Reply:
done